web/modules/custom/custom_register/src/Service/CustomeFormValidator.php

```
@@ -0,0 +1,281 @@
+<?php
+
+namespace Drupal\custom_register\Service;
+
+use Drupal\Core\Database\Connection;
+use Drupal\Core\StringTranslation\TranslationInterface;
+
+/**
+ * Provides the basic Custom Register form.
+ */
+class CustomeFormValidator {
+  /**
+   * Email validation.
+   *
+   * @var Drupal\Core\Mail\EmailValidatorInterface
+   */
+  protected $emailValidator;
+  /**
+   * The database connection.
+   *
+   * @var \Drupal\Core\Database\Connection
+   */
+  protected $database;
+
+  /**
+   * The string translation service.
+   *
+   * @var \Drupal\Core\StringTranslation\TranslationInterface
+   */
+  protected $stringTranslation;
+
+  /**
+   * {@inheritdoc}
+   */
+  public function __construct($email_validator, Connection $database, TranslationInterface $string_translation) {
+    $this->emailValidator = $email_validator;
+    $this->database = $database;
+    $this->stringTranslation = $string_translation;
+  }
+
+  /**
+   * Validates an email address.
+   *
+   * This method performs two checks:
+   *  1. Checks if the email format is valid using EmailValidator service.
+   *  2. Checks if the email already exists in the user database.
+   *
+   * @param string $email
+   *   The email address to validate.
+   *
+   * @return array
+   *   An array containing:
+   *    -[0]: A message string describing the validation result.
+   *    -[1]: A boolean indicating if the email is valid (TRUE) or not (FALSE).
+   */
+  public function getEmailValidationInfo($email) {
+    if (!$this->emailValidator->isValid($email)) {
+      return [
+        'is_valid' => FALSE,
+        'message' => $this->stringTranslation->translate('Invalid email format'),
+      ];
+    }
+
+    $existing_users = $this->database->select('custom_user_data', 'c')
+      ->fields('c', ['id'])
+      ->condition('email', $email)
+      ->execute()
+      ->fetchField();
+
+    if (empty($existing_users)) {
```

The variable always exists, so using `empty()` doesn't make sense here. Moreover, this can cause the interpreter to silently ignore errors about non-existent variables, and even some static analysis tools will consider the code valid even if the code above is changed and the variable no longer exists.